spotdl/download/downloader.py

```diff
@@ -24,6 +24,81 @@
 # === Base functionality ===
 # ==========================
 
+# ========================
+# === Helper function ===
+# ========================
+
+def _sanitize_filename(input_str: str) -> str:
+    # ! this is windows specific (disallowed chars)
+    output = input_str
+    for disallowed_char in ['/', '?', '\\', '*', '|', '<', '>']:
+        if disallowed_char in output:
+            output = output.replace(
+                disallowed_char,
+                ''
+            )
+
+    # ! double quotes (") and semi-colons (:) are also disallowed characters but we would
+    # ! like to retain their equivalents, so they aren't removed in the prior loop
+    output = output.replace('"', "'").replace(':', '-')
+
+    return output
+
+
+def _get_smaller_file_path(input_song: SongObj) -> Path:
+    # Only use the first artist if the song path turns out to be too long
+    smaller_name = f"{input_song.get_contributing_artists()[0]} - {input_song.get_song_name()}"
+
+    # this is windows specific (disallowed chars)
+    for disallowed_char in ['/', '?', '\\', '*', '|', '<', '>']:
+        if disallowed_char in smaller_name:
+            smaller_name = smaller_name.replace(disallowed_char, '')
+
+    # ! double quotes (") and semi-colons (:) are also disallowed characters
+    # ! but we would like to retain their equivalents, so they aren't removed
+    # ! in the prior loop
+    smaller_name = _sanitize_filename(smaller_name.replace('"', "'").replace(':', '-'))
+
+    try:
+        return Path(".", f"{smaller_name}.mp3").resolve()
+    except Exception:
+        raise OSError("Cannot save song due to path issues.")
+
+
+def _get_converted_file_path(song_obj: SongObj) -> Path:
+
+    # ! we eliminate contributing artist names that are also in the song name, else we
+    # ! would end up with things like 'Jetta, Mastubs - I'd love to change the world
+    # ! (Mastubs REMIX).mp3' which is kinda an odd file name.
+
+    # also make sure that main artist is included in artistStr even if they
+    # are in the song name, for example
+    # Lil Baby - Never Recover (Lil Baby & Gunna, Drake).mp3
+
+    artists_filtered = []
+
+    for artist in song_obj.get_contributing_artists():
+        if artist.lower() not in song_obj.get_song_name():
+            artists_filtered.append(artist)
+        elif artist.lower() is song_obj.get_contributing_artists()[0].lower():
+            artists_filtered.append(artist)
+
+    artist_str = ", ".join(artists_filtered)
+
+    converted_file_name = _sanitize_filename(f"{artist_str} - {song_obj.get_song_name()}.mp3")
+
+    converted_file_path = Path(".", f"{converted_file_name}")
```

Comment:
no need for the first dot `"."`, and the `converted_file_name` don't need to be in `f-string`

```suggestion
    converted_file_path = Path(converted_file_name)
```